[Bug]: Provider crashes when volume is removed

### What happened?

i used `terraform destroy` to remove resources. One of resources was volume. Provider did crash, but volume was removed.

```
│ The plugin6.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵

Stack trace from the terraform-provider-hcloud_v1.52.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x28 pc=0x10316efdc]

goroutine 42 [running]:
github.com/hetznercloud/terraform-provider-hcloud/internal/volume.resourceVolumeDelete({0x103607200, 0x14000257810}, 0x14000137a70, {0x1034b7a00?, 0x1400045b1e0})
        github.com/hetznercloud/terraform-provider-hcloud/internal/volume/resource.go:345 +0x28c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).delete(0x1400044e300, {0x103607158, 0x140002ff530}, 0x14000137a70, {0x1034b7a00, 0x1400045b1e0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.37.0/helper/schema/resource.go:900 +0xe4
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x1400044e300, {0x103607158, 0x140002ff530}, 0x140003881a0, 0x140001379e0, {0x1034b7a00, 0x1400045b1e0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.37.0/helper/schema/resource.go:949 +0x48c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x1400013c510, {0x103607158?, 0x140002ff3b0?}, 0x1400012fb30)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.37.0/helper/schema/grpc_provider.go:1484 +0xd48
github.com/hashicorp/terraform-plugin-mux/tf5to6server.v5tov6Server.ApplyResourceChange({{0x103613f98?, 0x1400013c510?}}, {0x103607158, 0x140002ff3b0}, 0x0?)
        github.com/hashicorp/terraform-plugin-mux@v0.20.0/tf5to6server/tf5to6server.go:38 +0x58
github.com/hashicorp/terraform-plugin-mux/tf6muxserver.(*muxServer).ApplyResourceChange(0x14000122480, {0x103607158?, 0x140002ff0e0?}, 0x1400012fae0)
        github.com/hashicorp/terraform-plugin-mux@v0.20.0/tf6muxserver/mux_server_ApplyResourceChange.go:36 +0x17c
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0x140001303c0, {0x103607158?, 0x140002fe5a0?}, 0x14000123780)
        github.com/hashicorp/terraform-plugin-go@v0.28.0/tfprotov6/tf6server/server.go:941 +0x294
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x1035d3940, 0x140001303c0}, {0x103607158, 0x140002fe5a0}, 0x14000123700, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.28.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:687 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x1400029a200, {0x103607158, 0x140002fe510}, 0x140002c8f60, 0x14000297860, 0x103d586c8, 0x0)
        google.golang.org/grpc@v1.72.1/server.go:1405 +0xc9c
google.golang.org/grpc.(*Server).handleStream(0x1400029a200, {0x103607af0, 0x1400045a1a0}, 0x140002c8f60)
        google.golang.org/grpc@v1.72.1/server.go:1815 +0x900
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.72.1/server.go:1035 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 35
        google.golang.org/grpc@v1.72.1/server.go:1046 +0x138

Error: The terraform-provider-hcloud_v1.52.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### What did you expect to happen?

Volume to be removed without errors

### Please provide a minimal working example

```
resource "hcloud_volume" "volume" {
  name              = var.name
  size              = var.size
  location          = var.location
  delete_protection = var.delete_protection
}
```

--- a/internal/volume/resource.go
+++ b/internal/volume/resource.go
@@ -341,6 +341,11 @@ func resourceVolumeDelete(ctx context.Context, d *schema.ResourceData, m interfa
 	if err != nil {
 		return hcloudutil.ErrorToDiag(err)
 	}
+	if volume == nil {
+		log.Printf("[WARN] volume (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
 
 	if volume.Server != nil {
 		err := control.Retry(control.DefaultRetries, func() error {

